crates/ntx-builder/src/state/mod.rs

```diff
@@ -116,13 +116,14 @@ impl State {
 
         let notes = state.store.get_unconsumed_network_notes().await?;
         for note in notes {
-            let prefix = note.account_prefix();
-
-            // Ignore notes which don't target an existing account.
-            let Some(account) = state.fetch_account(prefix).await? else {
-                continue;
-            };
-            account.add_note(note);
+            // TODO: support multi target network notes.
+            if let Some(prefix) = note.account_prefix() {
+                // Ignore notes which don't target an existing account.
+                let Some(account) = state.fetch_account(prefix).await? else {
+                    continue;
+                };
+                account.add_note(note);
+            }
```

Comment:
I would probably replace the `TODO` with a simple comment explaining that we handle only single-target accounts for now.

Also, we can probably re-write this as:
```rust
if let Some(prefix) = note.account_prefix() {
    let Some(account) = state.fetch_account(prefix).await? {
        account.add_note(note);
    }
}
```

Lastly, I would expand the comment and explain under which circumstances we may get a single-target note that doesn't have a target account.